specs/use_value_attribute_to_locate_object_paths_in_etcd

```diff
@@ -0,0 +1,104 @@
+// vim: tw=79
+
+= Use 'value' attribute in objects to locate object path in etcd
+
+Tendrl API should use the 'value' attribute defined in the definitions yaml
+files to infer the path of the object in etcd. These paths are used to retrieve
+a single instance or a list of the defined object.
+
+== Problem description
+
+Tendrl API currently locate object paths based on certain assumptions like the
+object path is the plural of the object name etc. The assumption might not
+always be correct and hence the paths for retrieving single instance or a list
+of the objects should be picked up from the definitions yaml files under the
+object section with a attribute defined as 'value'.
+
+== Use Cases
+
+Using the configurations from the definitions yaml files for locating path for
+finding objects makes the API future proof and reduces assumptions or conventions which might break in certain situations.
+
+== Proposed change
+
+Tendrl API should use the configuration defined under objects 'value' in the
+defitnitions yaml file to locate the path for specific object. The path should
+be used to retrieve a single instance or a list of the object.
+
+=== Alternatives
+
+Use and ensure convention is followed across all object naming and paths
+in etcd.
+
+=== Data model impact:
+
+None
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The proposed change is a internal change and will not affect existing API's.
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+None
+
+=== Security impact:
+
+None
+
+=== Other end user impact:
+
+None
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+None
+
+
```

Comment:
impact on tendrl definitions needs to be mentioned here and probably the details about the changes